java-checks/src/main/java/org/sonar/java/checks/RedundantRecordMethodsCheck.java

```diff
@@ -0,0 +1,179 @@
+      .collect(Collectors.toSet());
+
+    for (Tree member : targetRecord.members()) {
+      if (member.is(Tree.Kind.CONSTRUCTOR)) {
+        checkConstructor((MethodTree) member, components);
+      } else if (member.is(Tree.Kind.METHOD)) {
+        checkMethod((MethodTree) member, components, componentNames);
+      }
+    }
+  }
+
+  private void checkConstructor(MethodTree constructor, List<Symbol.VariableSymbol> components) {
+    if (constructor.block().body().isEmpty() || onlyDoesSimpleAssignments(constructor, components)) {
+      reportIssue(constructor.simpleName(), "Remove this redundant constructor which is the same as a default one.");
+    }
+  }
+
+  private void checkMethod(MethodTree method, List<Symbol.VariableSymbol> components, Set<String> componentsByName) {
+    String methodName = method.symbol().name();
+    if (!componentsByName.contains(methodName)) {
+      return;
+    }
+    if (onlyReturnsRawValue(method, components)) {
+      reportIssue(method.simpleName(), "Remove this redundant method which is the same as a default one.");
+    }
+  }
+
+  public static boolean onlyReturnsRawValue(MethodTree method, Collection<Symbol.VariableSymbol> components) {
+    Optional<ReturnStatementTree> returnStatement = getFirstReturnStatement(method);
+    if (!returnStatement.isPresent()) {
+      return false;
+    }
+    ExpressionTree expression = returnStatement.get().expression();
+    Symbol identifierSymbol;
+    if (expression.is(Tree.Kind.IDENTIFIER)) {
+      identifierSymbol = ((IdentifierTree) expression).symbol();
+    } else if (expression.is(Tree.Kind.MEMBER_SELECT)) {
+      identifierSymbol = (((MemberSelectExpressionTree) expression).identifier()).symbol();
+    } else {
+      return false;
+    }
+    return components.stream().anyMatch(identifierSymbol::equals);
+  }
+
+  private static Optional<ReturnStatementTree> getFirstReturnStatement(MethodTree method) {
+    return method.block().body().stream()
+      .filter(statement -> statement.is(Tree.Kind.RETURN_STATEMENT))
+      .map(ReturnStatementTree.class::cast)
+      .findFirst();
+  }
+
+  public static boolean onlyDoesSimpleAssignments(MethodTree constructor, List<Symbol.VariableSymbol> components) {
+    if (constructor.parameters().size() != components.size()) {
+      return false;
+    }
+    List<Symbol.VariableSymbol> parameters = constructor.parameters().stream()
+      .map(parameter -> (Symbol.VariableSymbol) parameter.symbol())
+      .collect(Collectors.toList());
+    List<AssignmentExpressionTree> assignments = extractAssignments(constructor.block().body());
+    Set<Symbol> componentsAssignedInConstructor = new HashSet<>();
+    for (AssignmentExpressionTree assignment : assignments) {
+      Optional<Symbol.VariableSymbol> variableSymbol = assignsParameterToComponent(assignment, components, parameters);
+      if (variableSymbol.isPresent()) {
+        componentsAssignedInConstructor.add(variableSymbol.get());
+      }
```

Comment:
I think this can be rewritten even simpler as:
```java
assignsParameterToComponent(assignment, components, parameters)
  .ifPresent(componentsAssignedInConstructor::add);
```